examples/gno.land/p/demo/merkle/merkle.gno

```diff
@@ -0,0 +1,129 @@
+package merkle
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+)
+
+type Hashable interface {
+	Bytes() []byte
+}
+
+type nodes []Node
+
+type Node struct {
+	hash []byte
+
+	position uint8
+}
+
+func (n Node) Hash() string {
+	return hex.EncodeToString(n.hash[:])
+}
+
+type Tree struct {
+	layers []nodes
+}
+
+func (t *Tree) Root() string {
+	for _, l := range t.layers {
+		if len(l) == 1 {
+			return l[0].Hash()
+		}
+	}
+	return ""
+}
+
+func NewTree(data []Hashable) *Tree {
+	tree := &Tree{}
+
+	leaves := make([]Node, len(data))
+
+	for i, d := range data {
+		hash := sha256.Sum256(d.Bytes())
+		leaves[i] = Node{hash: hash[:]}
+	}
+
+	tree.layers = []nodes{nodes(leaves)}
+
+	for len(leaves) > 1 {
+		level := make([]Node, 0, len(leaves)/2+1)
+		for i := 0; i < len(leaves); i += 2 {
+			var buff bytes.Buffer
```

Comment:
```suggestion

	var buff bytes.Buffer
	for len(leaves) > 1 {
		level := make([]Node, 0, len(leaves)/2+1)
		for i := 0; i < len(leaves); i += 2 {
			buff.Reset()
```

Reply:
Hmm i understand this change, but for readability, i this it's simpler to keep the buff variable under the scope of the loop ?

I'm curious in term of memory management if there is any differences... 🤔 

wdyt?

Reply:
Good to know, i just fixed it :)